File: kotlin_projects/03-1-login_kt2/app/src/main/java/com/example/login_kt2/repository/room/LoginTryDao.kt
package com.example.login_kt2.repository.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface LoginTryDao {

    @Insert
    suspend fun insert(loginTry : LoginTry)

    @Query("Select * from LoginTry Order by id DESC")
    suspend fun getAll() : List<LoginTry>

    @Query("Select * from LoginTry Order by id DESC")
    fun getAllInFlow() : Flow<List<LoginTry>>

}

File: flutter_projects/03-1-login_fl/lib/repository/floor/login_try_dao.dart

import 'package:floor/floor.dart';
import 'package:login_fl/repository/floor/login_try.dart';

@dao
abstract class LoginTryDao {

  @Insert(onConflict: OnConflictStrategy.abort)
  Future<void> insert(LoginTry loginTry);

  @Query('SELECT * FROM LoginTry ORDER BY id DESC')
  Future<List<LoginTry>> getAll();

  @Query('SELECT * FROM LoginTry ORDER BY id DESC')
  Stream<List<LoginTry>> getAllInStream();

}